From 5424133233233153cc473f9d1c836ff25b6e5214 Mon Sep 17 00:00:00 2001
From: kueken <ohnemailadresse@arcor.de>
Date: Thu, 2 Jan 2020 22:02:27 +0100
Subject: [PATCH] Revert "Use getImageArch for sh4 instead of manual platform
 selection."

This reverts commit 078f45f96b160193afdc7e2e0f4c89913648ff5e.

# Conflicts:
#	IPTVPlayer/components/configextmovieplayer.py
#	IPTVPlayer/components/iptvconfigmenu.py
#	IPTVPlayer/version.py
---
 IPTVPlayer/components/configextmovieplayer.py | 28 +++++++++++++--------------
 IPTVPlayer/components/iptvconfigmenu.py       |  5 +++++
 IPTVPlayer/components/iptvplayerwidget.py     |  6 +++---
 IPTVPlayer/iptvdm/iptvdmui.py                 |  4 ++--
 IPTVPlayer/tools/iptvtools.py                 |  3 +--
 5 files changed, 25 insertions(+), 21 deletions(-)

diff --git a/IPTVPlayer/components/configextmovieplayer.py b/IPTVPlayer/components/configextmovieplayer.py
index b8fe1c3..7e08191 100644
--- a/IPTVPlayer/components/configextmovieplayer.py
+++ b/IPTVPlayer/components/configextmovieplayer.py
@@ -19,7 +19,6 @@ import skin
 from enigma import eLabel, getDesktop
 from Screens.MessageBox import MessageBox
 from Components.config import config, ConfigSelection, ConfigYesNo, Config, ConfigInteger, getConfigListEntry, configfile
-from boxbranding import getImageArch
 ###################################################
 config.plugins.iptvplayer.extplayer_summary = ConfigSelection(default = "yes", choices = [('auto', _('Auto')), ('yes', _('Yes')), ('no', _('No'))])
 config.plugins.iptvplayer.use_clear_iframe = ConfigYesNo(default = False)
@@ -298,18 +297,19 @@ class ConfigExtMoviePlayer(ConfigBaseWidget, ConfigExtMoviePlayerBase):
         if getDesktop(0).size().width() >= 800:
             list.append(getConfigListEntry(_("Info bar clock format"), config.plugins.iptvplayer.extplayer_infobanner_clockformat))
         list.append(getConfigListEntry(_("Create LCD/VFD summary screen"), config.plugins.iptvplayer.extplayer_summary))
-        list.append(getConfigListEntry(_("----------------- External exteplayer3 options -----------------"), config.plugins.iptvplayer.fakeExtePlayer3))
-        list.append(getConfigListEntry("    " + _("RAM buffer size [MB] for network protocols"), config.plugins.iptvplayer.rambuffer_sizemb_network_proto))
-        list.append(getConfigListEntry("    " + _("RAM buffer size [MB] for local files"), config.plugins.iptvplayer.rambuffer_sizemb_files))
-        list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'AAC', config.plugins.iptvplayer.aac_software_decode))
-        list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'AC3', config.plugins.iptvplayer.ac3_software_decode))
-        list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'EAC3', config.plugins.iptvplayer.eac3_software_decode))
-        list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'DTS', config.plugins.iptvplayer.dts_software_decode))
-        list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'WMA', config.plugins.iptvplayer.wma_software_decode))
-        list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'MP3', config.plugins.iptvplayer.mp3_software_decode))
-        if getImageArch() != "sh4":
-            list.append(getConfigListEntry("    " + _("Software decoding as"), config.plugins.iptvplayer.software_decode_as))
-        list.append(getConfigListEntry("    " + _("Stereo downmix mode for software decoder"), config.plugins.iptvplayer.stereo_software_decode))
+	if 1:#IsExecutable(config.plugins.iptvplayer.exteplayer3path.value):
+		list.append(getConfigListEntry(_("----------------- External exteplayer3 options -----------------"), config.plugins.iptvplayer.fakeExtePlayer3))
+	list.append(getConfigListEntry("    " + _("RAM buffer size [MB] for network protocols"), config.plugins.iptvplayer.rambuffer_sizemb_network_proto))
+	list.append(getConfigListEntry("    " + _("RAM buffer size [MB] for local files"), config.plugins.iptvplayer.rambuffer_sizemb_files))
+	list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'AAC', config.plugins.iptvplayer.aac_software_decode))
+	list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'AC3', config.plugins.iptvplayer.ac3_software_decode))
+	list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'EAC3', config.plugins.iptvplayer.eac3_software_decode))
+	list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'DTS', config.plugins.iptvplayer.dts_software_decode))
+	list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'WMA', config.plugins.iptvplayer.wma_software_decode))
+	list.append(getConfigListEntry("    " + _("Use software decoder for the %s") % 'MP3', config.plugins.iptvplayer.mp3_software_decode))
+	if not config.plugins.iptvplayer.sh4platform.value:
+		list.append(getConfigListEntry("    " + _("Software decoding as"), config.plugins.iptvplayer.software_decode_as))
+	list.append(getConfigListEntry("    " + _("Stereo downmix mode for software decoder"), config.plugins.iptvplayer.stereo_software_decode))
         if self.ac3_mix_avaliable:
             list.append(getConfigListEntry(_("AC3 downmix mode"), config.plugins.iptvplayer.ac3_mix))
         if self.aac_mix_avaliable:
@@ -359,7 +359,7 @@ class ConfigExtMoviePlayer(ConfigBaseWidget, ConfigExtMoviePlayerBase):
         list.append(getConfigListEntry(_("Show iframe during audio playback"), config.plugins.iptvplayer.show_iframe))
         if config.plugins.iptvplayer.show_iframe.value:
             list.append(getConfigListEntry("    " + _("Radio iframe file"), config.plugins.iptvplayer.iframe_file))
-        if getImageArch() != "sh4" and (config.plugins.iptvplayer.show_iframe.value or config.plugins.iptvplayer.use_clear_iframe.value):
+        if not config.plugins.iptvplayer.sh4platform.value and (config.plugins.iptvplayer.show_iframe.value or config.plugins.iptvplayer.use_clear_iframe.value):
             list.append(getConfigListEntry("    " + _("Black iframe file"), config.plugins.iptvplayer.clear_iframe_file))
         self.list = list
         ConfigBaseWidget.runSetup(self)
diff --git a/IPTVPlayer/components/iptvconfigmenu.py b/IPTVPlayer/components/iptvconfigmenu.py
index 5a6707b..5a393db 100644
--- a/IPTVPlayer/components/iptvconfigmenu.py
+++ b/IPTVPlayer/components/iptvconfigmenu.py
@@ -36,6 +36,8 @@ from Plugins.Extensions.IPTVPlayer.components.configextmovieplayer import Config
 config.plugins.iptvplayer.hlsdl_download  = ConfigYesNo(default = True)
 config.plugins.iptvplayer.set_curr_title  = ConfigYesNo(default = False)
 config.plugins.iptvplayer.curr_title_file = ConfigText(default = "", fixed_size = False) 
+config.plugins.iptvplayer.sh4platform     = ConfigYesNo(default = False)
+config.plugins.iptvplayer.platformfpuabi  = ConfigSelection(default = "", choices = [("", ""),("hard_float", _("Hardware floating point")),("soft_float", _("Software floating point"))])
 
 config.plugins.iptvplayer.showcover          = ConfigYesNo(default = True)
 config.plugins.iptvplayer.deleteIcons        = ConfigSelection(default = "3", choices = [("0", _("after closing")),("1", _("after day")),("3", _("after three days")),("7", _("after a week"))]) 
@@ -266,6 +268,8 @@ class ConfigMenu(ConfigBaseWidget):
             list.append( getConfigListEntry(_("    Show suggestions"), config.plugins.iptvplayer.osk_allow_suggestions) )
             list.append( getConfigListEntry(_("    Default suggestions provider"), config.plugins.iptvplayer.osk_default_suggestions) )
 
+        list.append( getConfigListEntry(_("SH4 platform"), config.plugins.iptvplayer.sh4platform) )
+        list.append( getConfigListEntry(_("FPU"), config.plugins.iptvplayer.platformfpuabi) )
         list.append( getConfigListEntry(_("Services configuration"), config.plugins.iptvplayer.fakeHostsList) )
         list.append( getConfigListEntry(_("Remove disabled services"), config.plugins.iptvplayer.remove_diabled_hosts) )
         list.append( getConfigListEntry(_("Initialize web interface (experimental)"), config.plugins.iptvplayer.IPTVWebIterface) )
@@ -438,6 +442,7 @@ class ConfigMenu(ConfigBaseWidget):
               config.plugins.iptvplayer.ListaGraficzna,
               config.plugins.iptvplayer.pluginProtectedByPin,
               config.plugins.iptvplayer.configProtectedByPin,
+              config.plugins.iptvplayer.sh4platform,
               config.plugins.iptvplayer.osk_type,
               config.plugins.iptvplayer.preferredupdateserver,
               ]
diff --git a/IPTVPlayer/components/iptvplayerwidget.py b/IPTVPlayer/components/iptvplayerwidget.py
index c5e1849..13efb83 100644
--- a/IPTVPlayer/components/iptvplayerwidget.py
+++ b/IPTVPlayer/components/iptvplayerwidget.py
@@ -27,7 +27,7 @@ from Tools.BoundFunction import boundFunction
 from Tools.LoadPixmap import LoadPixmap
 from Tools.Directories import fileExists
 from enigma import getDesktop, eTimer
-from boxbranding import getImageArch
+
 ####################################################
 #                   IPTV components
 ####################################################
@@ -1717,7 +1717,7 @@ class E2iPlayerWidget(Screen):
                     gstAdditionalParams['show_iframe'] = config.plugins.iptvplayer.show_iframe.value
                     gstAdditionalParams['iframe_file_start'] = config.plugins.iptvplayer.iframe_file.value
                     gstAdditionalParams['iframe_file_end'] = config.plugins.iptvplayer.clear_iframe_file.value
-                    if getImageArch() == "sh4":
+			if config.plugins.iptvplayer.sh4platform.value:
                         gstAdditionalParams['iframe_continue'] = True
                     else:
                         gstAdditionalParams['iframe_continue'] = False
@@ -1739,7 +1739,7 @@ class E2iPlayerWidget(Screen):
                             playerVal = 'gstplayer'
                             gstAdditionalParams['download-buffer-path'] = ''
                             gstAdditionalParams['ring-buffer-max-size'] = 0
-                            if getImageArch() == "sh4": # use default value, due to small amount of RAM
+				if config.plugins.iptvplayer.sh4platform.value: # use default value, due to small amount of RAM
                                 #use the default value, due to small amount of RAM
                                 #in the future it will be configurable
                                 gstAdditionalParams['buffer-duration'] = -1
diff --git a/IPTVPlayer/iptvdm/iptvdmui.py b/IPTVPlayer/iptvdm/iptvdmui.py
index ee8dbd7..9cdab5d 100644
--- a/IPTVPlayer/iptvdm/iptvdmui.py
+++ b/IPTVPlayer/iptvdm/iptvdmui.py
@@ -28,8 +28,8 @@ from Components.config import config
 from Components.ActionMap import ActionMap
 from Components.Label import Label
 from Components.Sources.StaticText import StaticText
+
 from os import chmod as os_chmod, path as os_path, remove as os_remove
-from boxbranding import getImageArch
 ###################################################
 
 #########################################################
@@ -313,7 +313,7 @@ class IPTVDMWidget(Screen):
                         additionalParams['show_iframe'] = config.plugins.iptvplayer.show_iframe.value
                         additionalParams['iframe_file_start'] = config.plugins.iptvplayer.iframe_file.value
                         additionalParams['iframe_file_end'] = config.plugins.iptvplayer.clear_iframe_file.value
-                        if getImageArch() == "sh4":
+                        if config.plugins.iptvplayer.sh4platform.value:
                             additionalParams['iframe_continue'] = True
                         else:
                             additionalParams['iframe_continue'] = False
diff --git a/IPTVPlayer/tools/iptvtools.py b/IPTVPlayer/tools/iptvtools.py
index 210a58b..e193530 100644
--- a/IPTVPlayer/tools/iptvtools.py
+++ b/IPTVPlayer/tools/iptvtools.py
@@ -27,7 +27,6 @@ import os
 import stat
 import codecs
 import datetime
-from boxbranding import getImageArch
 
 SERVER_DOMAINS = {'vline':'http://iptvplayer.vline.pl/', 'gitlab':'http://zadmario.gitlab.io/'}
 SERVER_UPDATE_PATH = {'vline':'download/update2/', 'gitlab':'update2/'}
@@ -110,7 +109,7 @@ def GetNice(pid=None):
     return nice
 
 def E2PrioFix(cmd, factor=2):
-    if getImageArch() != "sh4":
+    if not config.plugins.iptvplayer.sh4platform.value:
         return 'nice -n %d %s' % (GetNice() + factor, cmd)
     else:
         return cmd
-- 
2.7.4

